internal/secrets/driver/driver.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package driver defines interfaces to be implemented for providers of the
+// secrets package.
+package driver
+
+import "context"
+
+// Decryptor decrypts a cipher message into a plain text message. The
+// implementation of the Decryptor should hold any key information with it, for
+// example, the key name.
+type Decryptor interface {
+
+	// Decrypt decrypts the ciphertext and returns the plaintext. It should handle
+	// any necessary base64 encoding/decoding based on the service provider
```

Comment:
This seems like the wrong level of detail for an interface. Base64 or other transport details are specific to that implementation.